### DISCUSSION TITLE
Custom HTML on Item Detail

### DISCUSSION THREAD

**Discussion by louiswol94** [directus/directus#4178] [Archive]
Good day,

Please advise on what is the best way to add custom HTML on the edit screen of a collection.

We have a collection called "projects" and we just want to display some HTML when a user is editing a project.

The HTML will need to contain variables from the current item that is being edited.

I just need to know if there is a correct way to do this before I start hacking away in a direction.

~~Using V8.8.1~~

Kind regards

**Comment by benhaynes** [comment]
Technically this should be transferred to https://github.com/directus/v8-archive since it's for v8, but I'll leave it here since it applies to v9 as well. 

Interfaces are the easiest way to get custom logic onto the Item Detail page. You can create a [custom interface](https://docs.directus.io/guides/interfaces/) to show whatever you want... but what you mentioned about referencing field values is interesting!

@Nitwel @rijkvanzanten — thoughts on making the existing "Notice" (presentation only) interface support template variables? I haven't thought through the use-case, but it would be formatted as: 

```
Please contact {{owner.name}} for any edits to this item that are required.
```

**Reply by benhaynes** [reply]
@louiswol94 — also, let me know if I am understanding the page you want this on correctly!

**Reply by rijkvanzanten** [reply]
Duplicate of https://github.com/directus/directus/discussions/3841 ?

**Reply by benhaynes** [reply]
Depends on the page, but maybe! I thought this was meant for Item Detail... and so extending the `Notice` interface would be a good _no-code_ way of doing it.

**Reply by louiswol94** [reply]
Hi gents,

Thanks for the reply!

@benhaynes you understood correctly. It is in the item detail page. So when editing an item.

I do need some variables in my custom HTML block...

I will look into the custom interfaces so long.

Thanks so much.

**Reply by philleepflorence** [reply]
@louiswol94 - For Directus 8, consider using or [extending](https://v8.docs.directus.io/extensions/interfaces.html) the [Divider Interface](https://v8.docs.directus.io/guides/interfaces.html#divider), which currently allows you to enter simple Markdown (HTML) in the **Note** input under **Schema** in the [Field/Interface Settings](https://v8.docs.directus.io/guides/interfaces.html#choosing-an-interface).

> Hint
> To read data from an existing item in a custom interface - `this.values`.
> You can access other props using [Extension Toolkit - Mixins - Interface.js](https://github.com/directus/v8-archive/blob/master/extension-toolkit/mixins/interface.js)

**Comment by louiswol94** [comment]
I get this error when trying to add a custom interface when running 'npx rollup -c'

Vue packages version mismatch:

- vue@3.0.5 (/var/directus/src/node_modules/vue/index.js)
- vue-template-compiler@2.6.12 (/var/directus/src/node_modules/vue-template-compiler/package.json)

**Reply by louiswol94** [reply]
2.6.12 is the latest available version for vue-template-compiler
But this means I have to use an older version of "vue"?

**Comment by qddtdlko** [comment]
do u mean 
https://github.com/directus/directus/discussions/12634#discussioncomment-2645159
